Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterDetail.java

```diff
@@ -391,20 +390,17 @@ private IHDIStorageAccount getDefaultStorageAccount(Map<String, String> coresite
             throw new HDIException("Failed to get default storage root path");
         }
 
-        StoragePathInfo pathInfo = new StoragePathInfo(defaultStorageRootPath);
-        switch (pathInfo.storageType) {
+        StorageAccountType storageType = StorageAccountType.parseUri(URI.create(defaultStorageRootPath));
+        switch (storageType) {
             case ADLS:
                 return new ADLSStorageAccount(this, true, clusterIdentity, URI.create(defaultStorageRootPath));
 
             case BLOB:
-                String storageAccountName = pathInfo.path.getHost();
-                if (StringUtils.isBlank(storageAccountName)) {
-                    throw new HDIException(String.format("Failed to get default storage account name from root path %s with %s storage type",
-                            defaultStorageRootPath, pathInfo.storageType));
-                }
-
-                String defaultContainerName = pathInfo.path.getUserInfo();
-                String defaultStorageAccountKey = StorageAccountKeyPrefix + storageAccountName;
+                WasbUri wasbUri = WasbUri.parse(defaultStorageRootPath);
+                String storageAccountName = wasbUri.getStorageAccount();
+                String defaultContainerName = wasbUri.getContainer();
+                String defaultStorageAccountKey =
+                        StorageAccountKeyPrefix + storageAccountName + ".blob." + wasbUri.getEndpointSuffix();
```

Comment:
Maybe we can change the regex pattern in `WasbUri` to include `.blob.` in the endpointSuffix field, therefore here the code could be `StorageAccountKeyPrefix + storageAccountName + wasbUri.getEndpointSuffix()`